component-tools/src/main/java/org/talend/sdk/component/tools/DitaDocumentationGenerator.java

```diff
@@ -98,64 +100,128 @@ protected String emptyDefaultValue() {
+
+        generateConfigurationSection(xml, body, family, name, reference.getAttribute("id"),
+                parametersWithUInfo2.get("datastore"), "Datastore");
+        generateConfigurationSection(xml, body, family, name, reference.getAttribute("id"),
+                parametersWithUInfo2.get("dataset"), "Dataset");
+        generateConfigurationSection(xml, body, family, name, reference.getAttribute("id"),
+                parametersWithUInfo2.get(""), "Other");
+
+        reference.appendChild(body);
+
+        final StringWriter writer = new StringWriter();
+        final StreamResult result = new StreamResult(writer);
+        try {
+            final Transformer transformer = transformerFactory.newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+            transformer.transform(new DOMSource(xml), result);
+
+            final String rootDir = output.getName().replace(".zip", "");
+            if (directories.add(rootDir)) {
+                zip.putNextEntry(new ZipEntry(rootDir + '/'));
+                zip.closeEntry();
+            }
+            final String ditaFolder = rootDir + '/' + family;
+            if (directories.add(ditaFolder)) {
+                zip.putNextEntry(new ZipEntry(ditaFolder + '/'));
+                zip.closeEntry();
+            }
+
+            final String path = ditaFolder + '/' + name + ".dita";
+            zip.putNextEntry(new ZipEntry(path));
+            final String content = writer.toString();
+            final int refIdx = content.indexOf("<reference");
+            zip
+                    .write((content.substring(0, refIdx)
+                            + "<!DOCTYPE reference PUBLIC \"-//Talend//DTD DITA Composite//EN\" \"TalendDitabase.dtd\">\n"
+                            + content.substring(refIdx)).getBytes(StandardCharsets.UTF_8));
+            zip.closeEntry();
+        } catch (final IOException | TransformerException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private void generateConfigurationSection(final Document xml, final Element body, final String family,
+            final String name, final String id, final Map<String, List<Param>> parameters, final String sectionName) {
+        if (parameters == null) {
+            return;
+        }
+
         final Element section = xml.createElement("section");
-        section.setAttribute("id", "section_" + componentId);
+        section.setAttribute("id", "section_" + id);
         section.setAttribute("outputclass", "subscription");
         body.appendChild(section);
 
         final Element sectionTitle = xml.createElement("title");
-        sectionTitle.setTextContent("Parameters for " + family + " " + name + " component.");
+        sectionTitle.setTextContent(sectionName + " parameters for " + family + " " + name + " component.");
```

Comment:
Do you want the family/component id or display name there? Seems you use the id.

Reply:
Same as before, I will recreate a ticket if need to change.